adapta/security/clients/aws/_aws_client.py

```diff
@@ -0,0 +1,92 @@
+"""
+ Amazon Web Services implementation of AuthenticationClient.
+"""
+#  Copyright (c) 2023. ECCO Sneaks & Data
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+from typing import Optional, Dict
+
+import boto3
+from pyarrow.filesystem import FileSystem
+
+from adapta.security.clients._base import AuthenticationClient
+from adapta.security.clients.aws._aws_credentials import AccessKeyCredentials, EnvironmentAwsCredentials
+from adapta.storage.models.base import DataPath
+
+
+class AwsClient(AuthenticationClient):
```

Comment:
```suggestion
@final
class AwsClient(AuthenticationClient):
```
because data scientists lol :)